api/v1beta1/openstackdataplane_webhook.go

```diff
@@ -66,6 +66,7 @@ func (spec *OpenStackDataPlaneSpec) Default() {
 			roleSSHSecret := spec.Roles[node.Role].NodeTemplate.AnsibleSSHPrivateKeySecret
 			node.Node.AnsibleSSHPrivateKeySecret = roleSSHSecret
 		}
+		spec.Nodes[nodeName] = *node.DeepCopy()
```

Comment:
Can we have the discussion around architecture first?

Under the current architecture, this makes sense. Particularly removing the NetworkConfig j2 file definitions. But do we want to change this setup around nodes in general is my broader question.